var/spack/repos/builtin/packages/pythia8/package.py

```diff
@@ -159,6 +159,8 @@ def configure_args(self):
         if self.spec.satisfies("+hdf5"):
             args.append("--with-highfive=" + self.spec["highfive"].prefix)
 
+        args += self.with_or_without("gzip", activation_value="prefix")
```

Comment:
Associate with the suggestion above:
```suggestion
        args += self.with_or_without(
            "gzip", activation_value=lambda x: self.spec["zlib-api"].prefix
        )
```